src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -41,21 +43,125 @@ export const Breadcrumbs = ({
   ariaLabel = 'breadcrumbs links',
   ...other
 }: Props) => {
+  const [shouldTruncate, setShouldTruncate] = React.useState(false);
+
+  const ref = React.useRef<HTMLUListElement>(null);
+
+  /**
+   * Checks if breadcrumbs would overflow
+   */
+  const updateShouldTruncate = () => {
+    setShouldTruncate(false);
+    if (
+      ref &&
+      ref.current &&
+      ref.current.clientWidth < ref.current.scrollWidth
+    ) {
+      setShouldTruncate(true);
+    }
+  };
+
+  /**
+   * Needs useLayoutEffect over useEffect since it needs to be run before paint.
+   * TODO: with React 18, might be able to use useEffect https://github.com/reactjs/reactjs.org/blob/d14cbdca2445cd676526c4c52e1e106342ff7bb3/content/docs/hooks-reference.md?plain=1#L155
+   */
+  React.useLayoutEffect(() => {
+    updateShouldTruncate();
+    window.addEventListener('resize', updateShouldTruncate);
```

Comment:
Should we debounce `updateShouldTruncate`?

Reply:
I was gonna ask the same thing. Seemed like the other window resize listeners in `Header`, `GlobalHeader` etc. were not debouncing and `lodash` isn't in our `package.json`. 

I do want to debounce this however, should I install `lodash.debounce` and do so?

Reply:
I think that's the right direction, yeah. I think in production it's weird to not have a debounce on something like this.